pkg/client/cache/watch_cache.go

```diff
@@ -27,12 +27,22 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
+// WatchCacheEvent is a single "watch event" that is send to users of
+// WatchCache. Additionally to a typical "watch.Event" it contains
+// the previous value of the object to enable proper filtering in the
+// upper layers.
```

Comment:
I like the idea of containing all the elements under storage so it is more self contained.